drivers/include/mlx90393.h

```diff
@@ -0,0 +1,403 @@
+    spi_clk_t clk;                      /**< clock speed for the SPI bus */
+#elif MODULE_MLX90393_I2C || DOXYGEN
+    i2c_t i2c;                          /**< I2C device */
+    uint8_t addr;                       /**< Magnometer I2C address */
+#endif
+    mlx90393_mode_t mode;                   /**< Measurement mode */
+    gpio_t int_pin;                         /**< Interrupt pin */
+    mlx90393_gain_t gain;                   /**< Analog chain gain */
+    mlx90393_resolution_t resolution;       /**< Desired 16-bit output value from the 19-bit ADC */
+    mlx90393_odr_t odr;                     /**< Output data rate */
+    mlx90393_oversampling_t oversampling;   /**< Oversampling ratio */
+    mlx90393_digital_filter_t dig_filt;     /**< Digital filter applicable to ADC */
+    mlx90393_treshold_t treshold;           /**< Treshold for wake-up on change mode */
+} mlx90393_params_t;
+
+/**
+ * @brief   Device descriptor for the MLX90393 magnetometer
+ */
+typedef struct {
+    mlx90393_params_t params;           /**< Device configuration parameters */
+    uint16_t ref_temp;                  /**< Reference temperature for converting raw temp data in centi celsius */
+    uint8_t conversion_time;            /**< Conversion time for single measurement mode */
+} mlx90393_t;
+
+/**
+ * @brief   Definition of error codes
+ * 
+ */
+typedef enum {
+    MLX90393_SUCCESS = 0,               /**< Success */
+    MLX90393_ERROR_I2C = -1,            /**< I2C Communication error */
+    MLX90393_ERROR_SPI = -2,            /**< SPI Communication error */
+    MLX90393_ERROR_NOT_AVAILABLE = -3,  /**< Device not available */
+    MLX90393_ERROR_NO_PIN = -4,         /**< No interrupt pin defined */
+    MLX90393_ERROR = -5,                /**< Device error */
+} mlx90393_error_t;
+
+/**
+ * @brief   Initialize the MLX90393 sensor device
+ * 
+ * This function resets the sensor and initializes it based on the given configuration 
+ * parameters. Burst and wake-up on change modes are activated immediately. In single 
+ * measurement mode, the measurement is not initiated by this function, mlx90393_read 
+ * must be called for that purpose. 
+ * If wake-up on change mode is used, configuring a valid interrupt pin is mandatory.
+ * 
+ * @note Some configurations of oversampling and digital filter are not permitted:
+ *       - OSR_0 and DIG_FILT_0
+ *       - OSR_0 and DIG_FILT_1
+ *       - OSR_1 and DIG_FILT_0
+ *       In such cases, the function will return with MLX90393_ERROR.
+ * 
+ * @note The conversion time increases with higher oversampling rates and digital filters.
+ *       If the values are too high, it may result in the configured output data rate 
+ *       not being achieved.
+ *
+ * @param[inout] dev        Device descriptor of MLX90393 device to be initialized
+ * @param[in]    params     Configuration parameters
+ *
+ * @retval MLX90393_SUCCESS     on success
+ * @retval MLX90393_SUCCESS     a negative error code, see mlx90393_error_t
```

Comment:
Es sollten die UNIX-Fehlercodes verwendet werden. Es macht sicher keinen Sinn, sämtliche I2C und SPI Error Codes im Detail durchzureichen. Es würde reichen die I2C und SPI Fehlercodes unter `-EIO` zusammen zu fassen.
```suggestion
 * @retval 0        on success
 * @retval -ENXIO   device not found
 * @retval -EIO     SPI or I2C communication error
 * @retval -EINVAL  invalid parameter
 * @retval -EFAULT  device error
```